etcd-cfn-signal/src/aws.rs

```diff
@@ -0,0 +1,153 @@
+extern crate hyper;
+extern crate rusoto;
+
+use std::str::FromStr;
+use std::io::Read;
+
+use hyper::Error;
+
+use rusoto::ec2;
+use rusoto::ec2::{Ec2Client, DescribeInstancesRequest, TagList};
+use rusoto::cloudformation::{CloudFormationClient, SignalResourceInput};
+use rusoto::{DefaultCredentialsProvider, Region};
+
+
+const INSTANCE_METADATA_URL: &'static str = "http://169.254.169.254/latest/meta-data/";
+const STACK_ID_TAG: &'static str = "aws:cloudformation:stack-id";
+const RESOURCE_ID_TAG: &'static str = "aws:cloudformation:logical-id";
+
+
+/// Signals cloudformation that we are ready.
+///
+/// Panics on failure.
+pub fn signal_cfn() {
+    temp();
```

Comment:
Actually that shouldn't even be there: debugging stuff that I removed in a future commit (but that commit is in a future PR).